R/test_trace_df.R

```diff
@@ -135,26 +135,63 @@ coverage_get_tests <- function(coverage) {
 
 
 
-#' Verify that the coverage object retains testing information
+#' Test that the coverage object retains testing information
 #'
 #' Test whether the coverage object has expected fields produced when coverage
 #' was captured with \code{option(covr.record_tests = TRUE)}.
 #'
 #' @param coverage a \code{\link[covr]{covr}} coverage object
 #' @family coverage_tests
 #'
-coverage_check_has_recorded_tests <- function(coverage) {
-  has_rec_tests <- !vapply(
-    coverage,
-    function(i) is.null(i[["tests"]]),
-    logical(1L))
+#' @importFrom utils getSrcDirectory
+#'
+coverage_has_recorded_tests <- function(coverage) {
+  has_tests_attr <- !is.null(attr(coverage, "tests"))
+
+  has_r_dir_traces <- !is.na(Position(
+    function(trace) basename(getSrcDirectory(trace$srcref)) == "R",
+    coverage
+  ))
+
+  has_trace_tests <- !is.na(Position(
+    function(trace) !is.null(trace[["tests"]]),
+    coverage
+  ))
+
+  if (has_tests_attr || has_trace_tests) return(TRUE)
+  else if (!has_r_dir_traces) return(NA)
+  else return(FALSE)
```

Comment:
Instead of checking and immediately throwing an error if traces weren't captured, this functionality was broken out into two separate functions. One that tests whether traces were captured (returning either `TRUE` (definitely), `FALSE` (definitely not), or `NA` (possibly) captured).

This result is passed to a function that then emits an error (when definitely not captured) or warning (possibly not captured) depending on severity. 

The warning should prompt a user to investigate whether it is expected that no R code is evaluated.